demo/src/screens/foundationScreens/ColorsScreen.js

```diff
@@ -1,115 +1,313 @@
 import _ from 'lodash';
 import React, {Component} from 'react';
-import {ScrollView, View, Text, StyleSheet, Dimensions, TouchableOpacity} from 'react-native';
-import {Colors, Typography, Toast} from 'react-native-ui-lib';//eslint-disable-line
+import {ScrollView, StyleSheet} from 'react-native';
+import {
+  Colors,
+  Spacings,
+  Assets,
+  View,
+  Text,
+  TouchableOpacity,
+  Icon,
+  Button,
+  TextField,
+  Incubator
+} from 'react-native-ui-lib';
 
+const {Toast} = Incubator;
 
-const {width} = Dimensions.get('window');
-const baseColors = ['grey', 'blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'purple', 'violet'];
+const SYSTEM_COLORS = ['grey', 'violet', 'blue', 'green', 'red', 'yellow', 'orange'];
+const INITIAL_COLOR = Colors.white;
+const BASE_PALETTE = ['1', '5', '10', '20', '30', '40', '50', '60', '70', '80'];
+const TOKENS_CATEGORIES = ['Background', 'Text', 'Icon', 'Outline'];
+const TOKENS_ARRAYS = {};
+TOKENS_CATEGORIES.map(category => (TOKENS_ARRAYS[category] = []));
 
-export default class ColorsScreen extends Component {
+for (const key in Colors) {
+  if (key.startsWith('$background')) {
+    TOKENS_ARRAYS.Background.push(key);
+  } else if (key.startsWith('$text')) {
+    TOKENS_ARRAYS.Text.push(key);
+  } else if (key.startsWith('$icon')) {
+    TOKENS_ARRAYS.Icon.push(key);
+  } else if (key.startsWith('$outline')) {
+    TOKENS_ARRAYS.Outline.push(key);
+  }
+}
 
+class ColorsScreen extends Component {
   constructor(props) {
     super(props);
 
+    this.extraColors = this.props.extraColors;
```

Comment:
if you're in the c'tor you don't need the 'this' before props, they are passed as a parameter. By the way, you don't need a c'tor at all as you can get the props using 'this' (as you did here).